core/thread.py

```diff
@@ -42,28 +42,58 @@ def ready(self, flag):
         if flag is True:
             self.ready_event.set()
     
-    async def _close_after(self, after, **kwargs):
-        await asyncio.sleep(after)
-        kwargs['scheduled'] = True
-        await self.close(**kwargs)
+    def _close_after(self, closer, silent, delete_channel, message):
+        return self.bot.loop.create_task(
+            self._close(closer, silent, delete_channel, message, True))
 
-    async def close(self, *, closer, after=0, silent=False, delete_channel=True, message=None, scheduled=False):
-        '''Close a thread now or after a set time in seconds'''
-        if self.close_task is not None and not self.close_task.cancelled():
-            if not scheduled or after > 0:
-                self.close_task.cancel()
+    async def close(self, *, closer, after=0, silent=False,
+                    delete_channel=True, message=None):
+        """Close a thread now or after a set time in seconds"""
+
+        if self.close_task is not None:
+            # restarts the after timer
+            self.close_task.cancel()
+            self.close_task = None
 
         if after > 0:
-            self.close_task = asyncio.create_task(self._close_after(after, closer=closer, silent=silent, message=message))
-            return 
+            # TODO: Add somewhere to clean up broken closures
+            #  (when channel is already deleted)
+            await self.bot.config.update()
+            now = datetime.datetime.utcnow()
+            items = {
+                # 'initiation_time': now.isoformat(),
+                'time': (now + datetime.timedelta(seconds=after)).isoformat(),
+                'closer_id': closer.id,
+                'silent': silent,
+                'delete_channel': delete_channel,
+                'message': message
+            }
+            self.bot.config.closures[str(self.id)] = items
+            await self.bot.config.update()
+
+            self.close_task = self.bot.loop.call_later(
+                after, self._close_after, closer,
+                silent, delete_channel, message)
+            return
 
+        return await self._close(closer, silent, delete_channel, message)
+
+    async def _close(self, closer, silent=False, delete_channel=True,
+                     message=None, scheduled=False):
         del self.manager.cache[self.id]
+
+        if scheduled:
+            self.bot.config.closures.pop(str(self.id), None)
+            await self.bot.config.update()
```

Comment:
This coro is already called further down in code, so there's no need to call it here.

Reply:
The referred code's outdated, don't believe there's anything to do with the current code.